yt_dlp/extractor/paramountplus.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from .cbs import CBSBaseIE
+from ..utils import (
+    int_or_none,
+    url_or_none,
+)
+
+
+class ParamountPlusIE(CBSBaseIE):
+    _VALID_URL = r'''(?x)
+        (?:
+            paramountplus:|
+            https?://(?:www\.)?(?:
+                paramountplus\.com/(?:shows/[^/]+/video|movies/[^/]+)/
+        )(?P<id>[\w-]+))'''
+
+    # All tests are blocked outside US
+    _TESTS = [{
+        'url': 'https://www.paramountplus.com/shows/catdog/video/Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k/catdog-climb-every-catdog-the-canine-mutiny/',
+        'info_dict': {
+            'id': 'Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k',
+            'ext': 'mp4',
+            'title': 'CatDog - Climb Every CatDog/The Canine Mutiny',
+            'description': 'md5:7ac835000645a69933df226940e3c859',
+            'duration': 1418,
+            'timestamp': 920264400,
+            'upload_date': '19990301',
+            'uploader': 'CBSI-NEW',
+        },
+        'params': {
+            'skip_download': 'm3u8',
+        },
+    }, {
+        'url': 'https://www.paramountplus.com/shows/tooning-out-the-news/video/6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd/7-23-21-week-in-review-rep-jahana-hayes-howard-fineman-sen-michael-bennet-sheera-frenkel-cecilia-kang-/',
+        'info_dict': {
+            'id': '6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd',
+            'ext': 'mp4',
+            'title': '7/23/21 WEEK IN REVIEW (Rep. Jahana Hayes/Howard Fineman/Sen. Michael Bennet/Sheera Frenkel & Cecilia Kang)',
+            'description': 'md5:f4adcea3e8b106192022e121f1565bae',
+            'duration': 2506,
+            'timestamp': 1627063200,
+            'upload_date': '20210723',
+            'uploader': 'CBSI-NEW',
+        },
+        'params': {
+            'skip_download': 'm3u8',
+        },
+    }, {
+        'url': 'https://www.paramountplus.com/movies/daddys-home/vM2vm0kE6vsS2U41VhMRKTOVHyQAr6pC',
+        'info_dict': {
+            'id': 'vM2vm0kE6vsS2U41VhMRKTOVHyQAr6pC',
+            'ext': 'mp4',
+            'title': 'Daddy\'s Home',
+            'upload_date': '20151225',
+            'description': 'md5:a0beaf24e8d3b0e81b2ee41d47c06f33',
+            'uploader': 'CBSI-NEW',
+            'timestamp': 1451030400,
+        },
+        'params': {
+            'skip_download': 'm3u8',
+            'format': 'bestvideo',
+        },
+        'expected_warnings': ['Ignoring subtitle tracks'],  # TODO: Investigate this
```

Comment:
We should extract these subs to `tp_subtitles` as well, but that's probably best left for a future PR since it requires editing `ThePlatformIE`